### DISCUSSION TITLE
Sort and filter bar chart

### DISCUSSION THREAD

**Discussion by dgwyer** [observablehq/plot#1027] [Q&A] [answered]
I found an example of a group and stacked bar chart and added them to my own notebook for practice [here](https://observablehq.com/@dgwyer/grouped-and-stacked-bar-chart). I'm still struggling with the syntax of Plot a little and am having difficulty sorting and filtering the two bar charts.

In the group bar chart, the results are sorted in descending order and filtered to show the first six facets. My questions are:

1. How would I do the same sort/filter for the stacked bar chart?
2. How could filter to show (for example) the first six data entries but not sort them?

**Comment by Fil** [comment, marked as answer]
For question 1, you could use the domain from scale fx in the group bar chart to define the scale x of the stacked bar chart? Like so;
```
x: {
  type: "band",
  tickFormat: d => d.toLocaleString("en", {month: "narrow"}),
  label: null,
  domain: d3.groupSort(crimea, v => d3.sum(v, d => -d.deaths), d => d.date).slice(0, 6),
}
```


For question 2, if you want the first six (different!) dates in input order, you could do:
`domain: [...new d3.InternSet(crimea.map(d => d.date))].slice(0, 6)` ?

**Comment by dgwyer** [comment]
Thanks that works great.
